app/models/chef.rb

```diff
@@ -1,4 +1,9 @@
 class Chef < ApplicationRecord
    validates_presence_of :name
    has_many :dishes
+   has_many :ingredients, through: :dishes
+
+   def self.unique_ingredients(chef_id)
+      joins(dishes: :ingredients).where(chefs: { id: chef_id }).select("ingredients.*").distinct
+   end
```

Comment:
This method, although correct, is doing way more than necessary. If this were an instance method, you could trim this method down to:

```ruby
def unique_ingredients
  ingredients.distinct # Making use of the already created association.
end
```